google-cloud-nio/src/main/resources/META-INF/native-image/com/google/cloud/google-cloud-nio/native-image.properties

```diff
@@ -0,0 +1,36 @@
+# Using META-INF/services with Native Image compilation results in
+# FileSystemProvider being initialized at build time. This results
+# CloudStorageFileSystemProvider and all the sub-classes referenced by
+# this class also being initialized at build time.
```

Comment:
Only for static fields or static initializers are needed.

Reply:
Slightly updated the comment here. Let me know if this sounds okay to you? Explored the code a little further to double check what's happening: The initialization of classes stems from `CloudStorageFileSystemProvider` which causes `StorageOptions` and as a result `StorageOptionsUtil`  to be initialized at build time.
https://github.com/googleapis/java-storage-nio/blob/004ab8c95dad31958701851381267e00e207da6a/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java#L96

Looks like this seems to be the case in the [reproducer](https://github.com/mpeddada1/junit-experiment) as well where when we create a new class `MyOptions` and instantiate it in `MySampleFileSystemProvider`, `MyOptions` ends up getting initialized at build time.